### ISSUE TITLE
Error logging

### ISSUE THREAD

**Issue by reddog** [healthlocker/healthlocker#1080]
There are no error logs for the HealthLocker app.  When run interactively from the console, error log information is displayed on-screen.

Error logging should be added in order to help locate and fix bugs.

**Comment by Danwhy** [comment]
We can use Elixir's [Logger](https://hexdocs.pm/logger/Logger.html) with [Logger File Backend](https://github.com/onkel-dirtus/logger_file_backend) for simple logging of errors to a file

**Comment by reddog** [comment]
@katbow Great to work with you and @Danwhy on Tuesday.

When we left I believe it was agreed that you would seek feedback from @SimonLab on your PR https://github.com/healthlocker/healthlocker/pull/1102 and then deploy the error logging to staging and live.  This is to support future investigations of #1101.  Please let me know if there are any issues.

**Comment by katbow** [comment]
I can't get the solution working locally. It doesn't seem to be logging anything.

**Comment by reddog** [comment]
@katbow Thanks for that, is that the same as on Tuesday?  Has @SimonLab been able to offer any suggestions?  Alternatively you suggested that it could be tested on staging?

When deploying (on staging or live), please do let us know so that we can check that the environment isn't currently in use for a demo, etc.

**Comment by reddog** [comment]
@katbow Could it be that nothing's being logged because there are no errors at that time?  Maybe re-try with a lower log level, or force an error?

**Comment by katbow** [comment]
I did force an error since that's what I needed to test. 

I'll see about getting it merged so I can deploy to staging and test there.

**Comment by reddog** [comment]
@katbow Thanks.  Were there any suggestions from @SimonLab?

Please let us know before deploying so we can at least alert others if not check that the server isn't being used for a demo or the like.

**Comment by katbow** [comment]
It's ready to deploy to staging now

**Comment by katbow** [comment]
Am I good to go on that @reddog ?

**Comment by reddog** [comment]
@katbow Can we give SLaM some notice?  Can we say that you'll be deploying at 1pm to the staging site?

**Comment by reddog** [comment]
Are you good with 1pm @katbow ?

**Comment by katbow** [comment]
I redeployed but it caused an error on staging. Looking at why now.

**Comment by reddog** [comment]
@katbow Was there an error message displayed?

**Comment by katbow** [comment]
No. It says it started with no errors

![image](https://user-images.githubusercontent.com/1287388/30966155-d8a8086c-a44f-11e7-8ab0-21789342876f.png)

But it's a 502 as if when it's not started

![image](https://user-images.githubusercontent.com/1287388/30966166-e8fad12c-a44f-11e7-8a80-af6a837052d5.png)

**Comment by reddog** [comment]
It's not to do with installing a new dependency, or creating the `healthlocker` folder in `/var/log` is it?

**Comment by reddog** [comment]
@katbow How are you getting on?  Should we revert to the previous, non-error logging version on staging?

**Comment by katbow** [comment]
yeah that's what I was checking out. I am trying from a fork to see if reverting fixes the issue.

**Comment by reddog** [comment]
@katbow Thanks for that.

**Comment by katbow** [comment]
Works with the reverted version. I'll make the change here and redeploy.
![image](https://user-images.githubusercontent.com/1287388/30971253-5369c20c-a45f-11e7-8a1b-074bff04a083.png)

![image](https://user-images.githubusercontent.com/1287388/30971281-645919c8-a45f-11e7-80bd-a2feaae75db8.png)

**Comment by reddog** [comment]
@katbow Thanks very much for that.  Does it need re-deploying again?

Were you able to ascertain if it is was anything to do with missing dependencies or the creation of the `/var/log/healthlocker` folder?

**Comment by katbow** [comment]
the dependencies were all there. I tried with the folder already created and not created. 

For this
>  Does it need re-deploying again?

See [here](https://github.com/healthlocker/healthlocker/issues/1080#issuecomment-332848979) and [here](https://github.com/healthlocker/healthlocker/issues/1080#issuecomment-332846450). It was a temporary deploy from a fork, so yes it will need a redeploy as the change was not reverted on this repo.

**Comment by reddog** [comment]
@katbow Thanks for confirming on those items.

You deployed from a fork of the code from just before this change?  And then reverted the change and that is the version to be re-deployed?  In which case the code should be the same in each case, no?  Hence my query if a further re-deploy would add anything different.

**Comment by katbow** [comment]
I forked the repo -> reverted the merge commit which gets rid of all the error logging changes -> deployed successfully without the error logging. 

I reverted the commit here now and am about to redeploy to the previous version as that is what worked.

**Comment by katbow** [comment]
Plan for updating appsignal for error logging

![image](https://user-images.githubusercontent.com/1287388/31019049-594c9dee-a526-11e7-8f33-c5ed4ec3e1b4.png)

Link to changelog [here](https://github.com/appsignal/appsignal-elixir/blob/develop/CHANGELOG.md).

**Comment by reddog** [comment]
With regard to file error logging, we don't have enough information to understand why this isn't working.

Would it be useful to create a **very simple** Phoenix app with file logging enabled and test that?  Further details of configuration can be found above at https://github.com/healthlocker/healthlocker/issues/1080#issuecomment-332209541

**Comment by reddog** [comment]
At this stage, this issue should only be completed if it has not been possible to complete #1106 (within agreed timescale, etc).

**Comment by reddog** [comment]
Issue raised against https://github.com/onkel-dirtus/logger_file_backend for Phoenix installation notes at https://github.com/onkel-dirtus/logger_file_backend/issues/40  Suggests it's necessary to run

```
mix deps.get
```

after making config changes.

Does change in `mix,lock` suggest that this has been run?

**Comment by katbow** [comment]
yes, it's been run.

**Comment by katbow** [comment]
@katyedwards can you verify if this can be closed as no longer a requirement due to #1106 being completed?

**Comment by katyedwards** [comment]
Thanks @katbow - I'm going to backlog it as it might be something we want to consider for the future as I don't think the trust want to use AppSignal going forward.
